tests/model_hub_tests/models_hub_common/test_performance_model.py

```diff
@@ -124,22 +134,41 @@ def get_read_model(self, model_path: str):
         return core.read_model(model=model_path)
 
     def infer_model(self, ov_model, inputs):
+        results = ModelResults()
         infer_step_t0 = time.time()
         # heat run
-        for _ in range(0, TestModelPerformance.num_heat_runs):
-            ov_model(inputs)
-        # measure
-        results = []
-        for _ in range(0, TestModelPerformance.num_measure_runs):
+        left_time = float(runtime_heat_duration)
+        while left_time > 0:
             t0 = time.time()
             ov_model(inputs)
             t1 = time.time()
-            results.append(t1 - t0)
-        mean = np.mean(results)
-        var = np.std(results, ddof=1) * 100 / mean
+            left_time -= (t1 - t0)
+            results.heat_n_repeats += 1
+        print('heat done in {} repeats'.format(results.heat_n_repeats))
+        # measure
+        runtime_duration_ns = float(runtime_measure_duration) * (10 ** 9)
+        left_time = runtime_duration_ns
+        time_slices = []
+        while left_time > 0:
+            t0 = time.perf_counter_ns()
+            ov_model(inputs)
+            t1 = time.perf_counter_ns()
+            timedelta = t1 - t0
+            time_slices.append(timedelta)
+            left_time -= timedelta
+            results.infer_n_repeats += 1
```

Comment:
please put this performance measurement routine into separate function or wrapper. We will use it for other purpose to measure original FW performance